app/main.py

```diff
@@ -1,34 +1,104 @@
+        self.is_alive = False
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(self, start: tuple, end: tuple) -> None:
+        self.start = start
+        self.end = end
+        self.decks = self.create_decks()
+        self.is_drowned = False
 
-    def get_deck(self, row, column):
+    def __str__(self) -> str:
+        return f"Ship from {self.start} to {self.end}"
+
+    def create_decks(self) -> list[Deck]:
+        decks = []
+        for row in range(self.start[0], self.end[0] + 1):
+            for column in range(self.start[1], self.end[1] + 1):
+                decks.append(Deck(row, column))
+        return decks
+
+    def get_deck(self, row: int, column: int) -> Deck:
         # Find the corresponding deck in the list
-        pass
+        for deck in self.decks:
+            if deck.row == row and deck.column == column:
+                return deck
 
-    def fire(self, row, column):
+    def fire(self, row: int, column: int) -> None:
         # Change the `is_alive` status of the deck
         # And update the `is_drowned` value if it's needed
-        pass
+        deck = self.get_deck(row, column)
+        deck.hit()
+        if all(not deck.is_alive for deck in self.decks):
+            self.is_drowned = True
 
 
 class Battleship:
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list[tuple]) -> None:
+        self.ships = [Ship(start, end) for start, end in ships]
+        self.field = self.create_field()
+
+    def create_field(self) -> dict[tuple, Ship]:
+        # Create a field with the ships
+        field = {}
+        for ship in self.ships:
+            for deck in ship.decks:
+                field[(deck.row, deck.column)] = ship
+        return field
+
+    def fire(self, location: tuple) -> str:
+        # Fire at the location
+        row, column = location
+        if (row, column) in self.field:
+            self.field[(row, column)].fire(row, column)
+            if self.field[(row, column)].is_drowned:
+                return "Sunk!"
+            return "Hit!"
+        return "Miss!"
+
+    def print_field(self) -> None:
+        # Print the field with the ships
+        for row in range(10):
+            for column in range(10):
+                if (row, column) in self.field:
+                    print(u"\u25A1", end=" ")
+                else:
+                    print("~", end=" ")
+            print()
```

Comment:
what if ship is Sunk or hitt?

Reply:
oh its need think about it. yes I wanted realize it)